independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/event/select/EventSelectTest.java

```diff
@@ -0,0 +1,87 @@
+package io.quarkus.arc.test.event.select;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.test.ArcTestContainer;
+import javax.enterprise.util.AnnotationLiteral;
+import javax.enterprise.util.TypeLiteral;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+/**
+ * Tests that event selection throws exceptions under certain circumstances.
+ */
+public class EventSelectTest {
+
+    @RegisterExtension
+    public ArcTestContainer container = new ArcTestContainer(BreakInEvent.class, NotABindingType.class,
+            SecurityEvent.class, SecurityEvent_Illegal.class,
+            SecuritySensor.class, SystemTest.class);
+
+    @Test
```

Comment:
Ok, call me Mr. Old Fart, but what's wrong with the following syntax?
```java
    @Test
    public <T> void testEventSelectThrowsExceptionIfEventTypeHasTypeVariable() {
       try {
          SecuritySensor sensor = Arc.container().select(SecuritySensor.class).get();
          sensor.securityEvent.select(new TypeLiteral<SecurityEvent_Illegal<T>>() {});
          fail( "Event#select should throw IllegalArgumentException if the event uses type variable");
       } catch (IllegalArgumentException expected) {
       }       
    } 
```
I mean, it's only one method, no lamda or method reference needed and the logic is clear and simple. Does `Assertions.assertThrows()` provide some benefit?

Reply:
So, Mr. Old Fart, there is nothing wrong with your syntax if you ask me ;-)
In fact, I'd have written the same code as you but I copied most of the test content from Scott's code which was using this syntax.
I can change it to try-catch blocks.

Reply:
I'm gonna say

```java
assertThrows(() -> {
    ...
});
```

is so much better han

```java
try {
    ...
    fail();
} catch (ExpectedException ignored) {
}
```

but these old farts just can't help themselves :-)